### ISSUE TITLE
Contact color in conersation overview.

### ISSUE THREAD

**Issue by fooness** [WhisperSystems/Signal-iOS#1434]
Hello folks, I’d love to be able to change the color of the circular contact icon in the conversation overview. Like you can in Android. Thank you very much!

**Comment by paewel** [comment]
Hello,
I'd like to get involved in development for Signal iOS. If an issue is labeled `help wanted`, then it's settled that it can be implemented?

**Comment by michaelkirk** [comment]
> then it's settled that it can be implemented?

That's correct. Since this provides some interface change, we need to ensure we're on the same page design wise. Signal-Android, which already has this feature, would be a good place to start.

Color palette is defined at https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/UIColor+OWS.m#L70

**Comment by paewel** [comment]
Can I find designs for android somewhere? Or some screenshots at least. Do I need to install android studio to see how it looks there? I don't own an Android phone unfortunately. I would be glad if someone posted me a screenshot here.

**Comment by sauberfred** [comment]
There are some screenshots on the Play Store page of Signal for Android.
https://play.google.com/store/apps/details?id=org.thoughtcrime.securesms

**Comment by naomihimley** [comment]
So my understanding of this is that on Android in the upper right hand corner in addition to the phone call icon which we have on iOS it also shows 3 dots that go to conversation settings. 

<img width="144" alt="upperrighthandcorner" src="https://cloud.githubusercontent.com/assets/6947424/20283034/a3713b8a-aa74-11e6-9c94-29101516e621.png">

Conversation settings on Android has many conversation specific settings such as Mute conversation, Notification sound, vibrate, Block, etc, as well as the Color (Color for this contact) option that opens a little color picker modal. 

Convo Settings Android:
<img width="440" alt="convosettingsandroid" src="https://cloud.githubusercontent.com/assets/6947424/20283067/c02ca7be-aa74-11e6-8764-f63f1f5c1851.png">

Color Picker Modal Android:
<img width="438" alt="colorpickerandroid" src="https://cloud.githubusercontent.com/assets/6947424/20283117/f1c4e732-aa74-11e6-810e-5b0ff8c0b76e.png">

Since the only design direction mentioned is to copy android, should we add the 3 dots that open a Conversation Settings screen and then have 1 option for Color (Color for this contact) that opens a color picker modal, and we can see about the other options in the future or is there discussion about Conversation Settings elsewhere?

**Comment by michaelkirk** [comment]
Thanks for the screenshots @naomihimley. On iOS currently, conversation settings are accessible via tapping on the contacts name (there's currently much less there than on Android).

Finding conversation settings [needs to be more obvious](https://github.com/WhisperSystems/Signal-iOS/issues/1390), but I would leave that for a separate PR.

Otherwise, I think that flow looks good. Reminder that our iOS color palette is different from Android and can be found here: https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/UIColor+OWS.m#L70

**Comment by naomihimley** [comment]
Ooh yep never have been into Conversation Settings. Ok that does bring up one other question to me now. On iOS we have icons on the left side of the cell. Android doesn't have that, but on the android cell they show a circle on the right side of the cell with the current selected color for this contact. For iOS do we want a new icon for color selection or how does a circle on the left side of the cell with the current color sound? @michaelkirk 

iOS example:
<img width="432" alt="iosconvosettings" src="https://cloud.githubusercontent.com/assets/6947424/20312595/4dc22256-ab11-11e6-974d-4b9c4970f9d0.png">

**Comment by michaelkirk** [comment]
> For iOS do we want a new icon for color selection or how does a circle on the left side of the cell with the current color sound?

I'm actually not sure. The icon on the left would probably be pretty redundant informationally, but aesthetically it might be weird against our otherwise monochromatic icons.

A mockup might clear it up. Thanks for putting forward some design proposals @naomihimley

**Comment by naomihimley** [comment]
Yep, happy to build it, I'm not a designer, but if we get an icon let me know.

**Comment by michaelkirk** [comment]
> Yep, happy to build it, I'm not a designer, but if we get an icon let me know.

I just mean happy that you're thinking about it first. 

For icon's I've been using https://material.io/icons/

e.g. https://material.io/icons/#ic_palette

They are unfortunately heavy stroked. I'm open to alternatives.

**Comment by abcdev** [comment]
I used [https://material.io/icons/#ic_brush](https://material.io/icons/#ic_brush) for this mockup because it felt more lightweight then the palette.  For the actual selection a UIPopoverPresentationController could be used with the view inside similar to the Android color picker above. But it would also look nice to slide in (from the top) a new cell presenting the available colors.
![contactcolor1](https://cloud.githubusercontent.com/assets/849672/20886476/7074d2c2-baf6-11e6-97a7-5a009062a957.png)


Differing from the monochromatic icons is somewhat strange.
![contactcolor2](https://cloud.githubusercontent.com/assets/849672/20886483/7551f5ae-baf6-11e6-9f10-98fa955b07ef.png)


IMHO omitting the icons looks somewhat cleaner. + It is closer to the Android version
![contactsettings](https://cloud.githubusercontent.com/assets/849672/20886485/7a1d4994-baf6-11e6-8302-00cb4c92c167.png)

**Comment by michaelkirk** [comment]
> IMHO omitting the icons looks somewhat cleaner. + It is closer to the Android version

I agree that it looks cleaner, but the icons do serve a purpose. In particular, the iconography for the disappearing messages,  the stopwatch, is the same symbols that you see when using this feature in the conversation view.

E.g. Say you have no idea that this feature exists. When the other person enables disappearing messages, you will see the clock appear in your conversation navbar as a mode indicator.

"What's this clock?"
Tap on it.
Be taken to settings.
See the clock next to the "disappearing messages" settings

> For the actual selection a UIPopoverPresentationController could be used with the view inside similar to the Android color picker above.

I agree that popover would be appropriate for the color picker.

Thanks for clearing the waters with these mocks @abcdev.

**Comment by sjauld** [comment]
Anyone working on this bad boy? IMHO it will sognificantly improve the group conversation UX.

**Comment by fooness** [comment]
+1

**Comment by michaelkirk** [comment]
It's on the roadmap, but not currently at the top of the list. Just follow this issue to get any progress updates.

Any implementation would need to sync the the changes to desktop, as android does when picking colors.

Also, please don't bump or +1 issues. It spams all repository subscribers with an email that says only and unhelpfully "+1". Repository subscribers by and large don't care that you "+1". 

We developers do care though. And you can more effectively show us which issues are important by using GH's built in reaction feature, which allows us to tally our most supported issues, all without spamming these poor souls.

**Comment by toothbrush** [comment]
Quite a few months ago i read that this feature was going to be coming back to iOS.  I'm curious if there's a timeline?  It's been a while without an update on this issue, and as someone who has switched from Android to iOS recently, i'm really missing my per-contact colours! Sorry to be a bother, i love your work 🙌

**Comment by herrfisk** [comment]
Any news on this feature?

**Comment by jessebett** [comment]
The lack of this feature is the main thing preventing me from buying an iphone.

**Comment by terencechow** [comment]
This is 100% needed. I've brought all my friends over and those with same initials is impossible to differentiate consider they haven't added pictures yet. Given this was last brought up in 2017, can we get some screenshots of existing Android so we can see what needs to be built for iOS to match Android?

**Comment by inokani** [comment]
Would these be of any help:
![Screenshot_20210109-121859_Signal_1](https://user-images.githubusercontent.com/77192679/104089381-3513ad80-5277-11eb-83dc-34c485ea7131.jpg)
![Screenshot_20210109-121933_Signal_2](https://user-images.githubusercontent.com/77192679/104089423-86bc3800-5277-11eb-9877-3e50ef93ce63.jpg)

**Comment by fooness** [comment]
I think many people would, since ever, like to leave Telegram, WhatsApp, et cetera in favor of Signal … if Signal would improve their interface in simple ways like this …

**Comment by rvaidun** [comment]
I have recently switched from Whatsapp to Signal. I hav iOS development experience and I would be happy to implement this in the app. Would I just open a PR when I implement it?

**Comment by bangsluke** [comment]
Is there an update on when IOS users will get coloured messages?

**Comment by Ge0rges** [comment]
@rvaidun i think you could just make a PR. We need this.

**Comment by Ectalite** [comment]
> It's on the roadmap, but not currently at the top of the list. Just follow this issue to get any progress updates.
> 
> Any implementation would need to sync the the changes to desktop, as android does when picking colors.
> 
> Also, please don't bump or +1 issues. It spams all repository subscribers with an email that says only and unhelpfully "+1". Repository subscribers by and large don't care that you "+1".
> 
> We developers do care though. And you can more effectively show us which issues are important by using GH's built in reaction feature, which allows us to tally our most supported issues, all without spamming these poor souls.

Is the desktop the issue ? If it is, you could add it in "beta" just for mobile on ios and when it is completely functionnal add the desktop part.

**Comment by Ectalite** [comment]
Still relevant.

**Comment by ohare93** [comment]
Still relevant, agreed.

**Comment by ryanprior** [comment]
This has been labeled "needs-design" since 2016. Probably correct just to close it as wontfix, Signal design is demonstrably not interested in touching this.

**Comment by ohare93** [comment]
Still relevant

**Comment by Cerberus0** [comment]
Please ignore the stale bot, it does not know the difference between bugs and feature requests because [this is no longer the right place to submit or discuss feature requests](https://github.com/signalapp/Signal-iOS/blob/main/CONTRIBUTING.md). Please let this GitHub issue be closed and continue this discussion on the community forum:

https://community.signalusers.org/t/add-color-code-for-each-contact/11167

Thanks!

**Comment by NiklasBr** [comment]
Still relevant
